src/factories/createSqlTag.ts

```diff
@@ -92,19 +96,62 @@ const sql: SqlTaggedTemplate = (
     }
   }
 
-  const query: SqlTokenType = {
+  return {
     sql: rawSql,
-    type: SqlToken,
     values: parameterValues,
   };
+};
+
+const sql = (
+  parts: readonly string[],
+  ...args: readonly ValueExpression[]
+) => {
+  const {
+    sql: sqlText,
+    values,
+  } = createQuery(parts, args);
+
+  const query = {
+    sql: sqlText,
+    type: SqlToken,
+    values,
+  };
 
   Object.defineProperty(query, 'sql', {
     configurable: false,
     enumerable: true,
     writable: false,
   });
 
-  return query;
+  return query as unknown as SqlSqlToken<QueryResultRow>;
+};
+
+sql.type = <T extends ZodTypeAny>(zodObject: T) => {
```

Comment:
What do you think about keeping it decoupled from zod, and just taking a parser in a shape that zod conforms to? Like 

```suggestion
interface Parser<T> {
  parse(input: unknown): T
}

sql.type = <T>(parser: Parser<T>) => {
```

No real benefit to explicitly coupling to zod I don't think.